pythonwars/pythonwars.py

```diff
@@ -42,13 +42,13 @@ def train_next_code_challenge(self, language, strategy="default", peek=False):
         }
         response = self.session.post(url, data=post_data)
         response.raise_for_status()
-        return CodeChallenge(response.json())
+        return TrainingCodeChallenge(response.json())
 
     def train_code_challenge(self, id_or_slug, language):
         url = _TRAIN_CODE_CHALLENGE_URL.format(id_or_slug, language)
         response = self.session.post(url)
         response.raise_for_status()
-        return CodeChallenge(response.json())
+        return TrainingCodeChallenge(response.json())
```

Comment:
this is starting to look familiar... a class method on `AttributesFromJson`, perhaps? (also should rename it at that point)